llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1528,6 +1531,38 @@ bool SPIRVToLLVM::postProcessRowMajorMatrix() {
         } else if (CallInst *const callInst = dyn_cast<CallInst>(value)) {
           if (callInst->getCalledFunction()->getName().startswith(gSPIRVMD::NonUniform))
             continue;
+        } else if (CmpInst *const cmpInst = dyn_cast<CmpInst>(value)) {
+          if (cmpInst->use_empty())
+            continue;
+
+          Value *lhs = nullptr;
+          Value *rhs = nullptr;
+          Value *mapOperand0 = nullptr;
+          Value *mapOperand1 = nullptr;
+
+          if (valueMap.count(cmpInst->getOperand(0)))
+            mapOperand0 = valueMap[cmpInst->getOperand(0)];
+          if (valueMap.count(cmpInst->getOperand(1)))
+            mapOperand1 = valueMap[cmpInst->getOperand(1)];
+
+          if (mapOperand0 && mapOperand1) {
+            lhs = mapOperand0;
+            rhs = mapOperand1;
+          } else if (mapOperand0) {
+            lhs = mapOperand0;
+            rhs = cmpInst->getOperand(1);
+          } else if (mapOperand1) {
+            lhs = cmpInst->getOperand(0);
+            rhs = mapOperand1;
+          } else {
+            llvm_unreachable("Should never be called!");
+          }
```

Comment:
It would be far more elegant to say something like:
```
Value *lhs = cmpInst->getOperand(0);
Value *rhs = cmpInst->getOperand(1);
if (Value *mapped = valueMap.lookup(lhs))
  lhs = mapped;
if (Value *mapped = valueMap.lookup(rhs))
  rhs = mapped;
```

Reply:
Yes, it looks way better ...